### PR TITLE
add JSON formatting to DataServlet

### PR DESCRIPTION
**Pull request by ncaa2018**
I created an arraylist of strings and had them parsed to JSON through gson. I also created a button that fetched the data from the /data URL server.

### CODE REVIEW

**Review by ncaa2018 — COMMENTED** [review]
added JSON for parsing info from the server and doPost for posting comments on the server. Still trying to figure out how to not automatically redirect to the server URl /data displaing "true" whenever I press "submit"

**Review by ncaa2018 — COMMENTED** [review]
added redirect back to /index.html

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -16,17 +16,54 @@
...
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Naomi!</h1>");
+    response.getWriter().println("<p>"+json+"</p>");
+
   }
+
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "comment", "");
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: rename text to newComment?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -16,17 +16,54 @@
...
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Naomi!</h1>");
+    response.getWriter().println("<p>"+json+"</p>");
+
   }
+
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nice commenting!

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -8,9 +8,18 @@
...
+      <p>Add comment:</p>
+      <textarea name="comment"></textarea>
+      <br/><br/>
+      <input type="submit" />
+      </form>
+
+     
+      <p>Click here to see all comments:</p>
+      <button onclick="getDataUsingAsyncAwait()">comments</button>
       <div id="data-container"></div>
```
Comment (renguyen) [inline portfolio/src/main/webapp/index.html]: nit: rename data-container to comments-container

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -8,9 +8,18 @@
...
+      <form action="/data" method="POST">
+      <p>Add comment:</p>
+      <textarea name="comment"></textarea>
+      <br/><br/>
+      <input type="submit" />
+      </form>
+
+     
+      <p>Click here to see all comments:</p>
+      <button onclick="getDataUsingAsyncAwait()">comments</button>
```
Comment (renguyen) [inline portfolio/src/main/webapp/index.html]: Just a note: this approach is totally fine and works, and if you wanted to be able to see the comments immediately without clicking a button, you can add `onload="getDataUsingAsyncAwait()"` to data-container below; see: https://www.w3schools.com/jsref/event_onload.asp

**Review by renguyen — COMMENTED** [review]
Great job! :) I also notice more details in your PR description, and I appreciate you taking my advice! I added a few comments to make your code even more specific (for example, text --> newComment, data-container --> comments-container). I would also build on your PR description and remember the "overview" aspect, so including something about how this PR is to add functionality for users to post comments and then see all posted comments as well. Overall, nice commenting and clean code!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,9 +34,70 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Naomi!</h1>");
+      //query to create some sort of order with the data
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit

```suggestion
      // Query to retrieve all comments, sorted by most recent first.
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -24,9 +34,70 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println("<p>"+json+"</p>");
+
+  }
+
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // //Arraylist to add comments in
+    // ArrayList<String> comments = new ArrayList<String>();
+
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]:
```suggestion
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -24,9 +34,70 @@
...
+    // // Get the input from the comment textbox
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]:
```suggestion
    // Get the input from the comment textbox.
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (renguyen) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]:
```suggestion
    String newComment = getParameter(request, "comment", "");
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -24,9 +34,70 @@
...
+    String text = getParameter(request, "comment", "");
+    long time = System.currentTimeMillis();
+
+    // // Respond with the result.
+    // response.setContentType("text/html;");
+    // response.getWriter().println(comments.add(text));
+
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]:
```suggestion
```

**Review by renguyen — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (ncaa2018) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: okay, doing this now

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (ncaa2018) [reply portfolio/src/main/webapp/index.html]: okay

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (ncaa2018) [reply portfolio/src/main/webapp/index.html]: mm okay, gotcha

**Review by ncaa2018 — COMMENTED** [review]
completed the nit suggestions by making changes tot he files

**Review by ncaa2018 — COMMENTED** [review]
created a comments class so I can loop through the comments stored in entities and put the in a list that is displayed.

**Review by ncaa2018 — COMMENTED** [review]
changed comments box to be bigger. Made comments load with text and date (formatted using java.util.Date in Comment.java class)

**Review by ncaa2018 — COMMENTED** [review]
moved comment section to the bottom and changed it from an onclick functionality to an onload

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -61,4 +53,24 @@ <h1>Naomi Amuzie's Portfolio</h1>
       <p>Click <a href="https://github.com/ncaa2018">here</a> for my GitHub profile. </p>
     </div>
   </body>
+
+
+    <body>
+    <div id="content">
```
Comment (renguyen) [inline portfolio/src/main/webapp/index.html:64]:
```suggestion
```

I think it makes more sense to have one <body><div id="content"></div></body> set.

File: portfolio/src/main/webapp/index.html [hunk 10]
```diff
@@ -61,4 +53,24 @@ <h1>Naomi Amuzie's Portfolio</h1>
...
+      <form action="/data" method="POST">
+      <p>Add comment:</p>
+      <textarea rows="5" cols="80" name="comment"></textarea>
+      <br/><br/>
+      <input type="submit" />
+      </form>
+
+      <p> Comments </p>
+      <body onload="getComments()">
+      <div id="comments-container"></div>
```
Comment (renguyen) [inline portfolio/src/main/webapp/index.html:76]:
```suggestion
      <div id="comments-container" onload="getComments()"></div>
```

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -32,45 +32,37 @@ function addRandomGreeting() {
...
-  // The fetch() function returns a Promise because the request is asynchronous.
-  const responsePromise = fetch('/data');
 
-  // When the request is complete, pass the response into handleResponse().
-  responsePromise.then(handleResponse);
+async function getComments() {
+  const response = await fetch('/data');
+  console.log('Fetching data from server.');
+  const comments = await response.json();
+  console.log(comments);
```
Comment (renguyen) [inline portfolio/src/main/webapp/script.js:42]: You might want to delete the extra console.log() statements to print comments and commentsElements (typically used for status updates, error printing, etc.)

File: portfolio/src/main/webapp/script.js [hunk 12]
```diff
@@ -32,45 +32,37 @@ function addRandomGreeting() {
...
+  const commentsElement = document.getElementById('comments-container'); //where I would like to post the comments that are fetched
+  console.log(commentsElement);
+  comments.forEach((comment) => {
+      commentsElement.appendChild(createCommentElement(comment));
```
Comment (renguyen) [inline portfolio/src/main/webapp/script.js:46]: Nice decomposition!

**Review by renguyen — APPROVED** [review]
Looks great! I really like the decomposition you did in creating the comment elements, and all your status updates on the PR conversation :) I added some comments about deleting extra console.log() statements and not having two sets of <body><div id="content"> and where to put the onload tag, but overall, great job with this complex feature!

**Pull request merged**